pandas/io/sql.py

```diff
@@ -2208,23 +2207,13 @@ def read_table(
         else:
             stmt = f"SELECT {select_list} FROM {table_name}"
 
-        mapping: type[ArrowDtype] | None | Callable
-        if dtype_backend == "pyarrow":
-            mapping = ArrowDtype
-        elif dtype_backend == "numpy_nullable":
-            from pandas.io._util import _arrow_dtype_mapping
-
-            mapping = _arrow_dtype_mapping().get
-        elif using_string_dtype():
-            from pandas.io._util import arrow_string_types_mapper
-
-            mapping = arrow_string_types_mapper()
-        else:
-            mapping = None
-
         with self.con.cursor() as cur:
             cur.execute(stmt)
-            df = cur.fetch_arrow_table().to_pandas(types_mapper=mapping)
+            pa_table = cur.fetch_arrow_table()
+            dtype_backend = (
+                lib.no_default if dtype_backend == "numpy" else dtype_backend
```

Comment:
It's also strange that we use "numpy" as the default in the SQL code, and `no_default` for all other places. 
Ideally we would solve that inconsistency as well, I think.

Now, I can certainly forward it and handle it inside `arrow_table_to_pandas` (I could also change the default here, and just additionally accept "numpy" for back compat)

Reply:
Hmm that seems like an issue in the SQL code. Looks like the read_sql API uses lib.no_default so that must be getting messed up in the internals. Let's keep this PR the way it is and track in https://github.com/pandas-dev/pandas/issues/60326

Reply:
Thanks for opening that issue.
Now, in the end I moved it to the utility anyway to deal with the typing issues (the type checkers don't understand that I removed "numpy" from the options ..)

Reply:
Sounds good. Still something to be cleaned up there, though not urgent for now